generators/service/languages/java/index.js

```diff
@@ -42,56 +36,24 @@ module.exports = class extends Generator {
 	initializing() {
 		this.context.languageFileExt = '';
 		this.context.generatorLocation = GENERATOR_LOCATION;
-		this.context.addMappings = this._addMappings.bind(this);
-		this.context.addLocalDevConfig = this._addLocalDevConfig.bind(this);
+		this.context.addMappings = ServiceUtils.addMappings.bind(this);
+		this.context.addLocalDevConfig = ServiceUtils.addLocalDevConfig.bind(this);
 		this.context.srcFolders = [];
 		this.context.instrumentationAdded = false;
 		this.context.metainf = [];
 		this._addJavaDependencies = Utils.addJavaDependencies.bind(this);
+		this.context.enable = ServiceUtils.enable.bind(this);
 
-		let serviceCredentials,
-			serviceKey;
-		//initializing ourselves by composing with the service enabler
-		let root = path.dirname(require.resolve('../../enabler'));
-		Object.keys(svcInfo).forEach(svc => {
-			serviceKey = svc;
-			serviceCredentials = this.context.application.service_credentials[serviceKey];
-			if (serviceCredentials) {
-				this.context.scaffolderKey = serviceKey;
-				logger.debug("Composing with service : " + svc);
-				try {
-					this.context.cloudLabel = serviceCredentials && serviceCredentials.serviceInfo && serviceCredentials.serviceInfo.cloudLabel;
-					this.composeWith(root, {context: this.context});
-				} catch (err) {
-					/* istanbul ignore next */	//ignore for code coverage as this is just a warning - if the service fails to load the subsequent service test will fail
-					logger.warn('Unable to compose with service', svc, err);
-				}
-			}
-		});
 	}
 
 	writing() {
 		// add missing pom.xml dependencies when running service enablement standalone
 		if (typeof this.context.parentContext === "undefined") {
 			this._addJavaDependencies();
 		}
+		this.context.enable()
 	}
-
-	_addMappings(serviceMappingsJSON) {
-		let mappingsFilePath = this.destinationPath(PATH_MAPPINGS_FILE);
-		this.fs.extendJSON(mappingsFilePath, serviceMappingsJSON);
-	}
-
-	_addLocalDevConfig(devconf) {
```

Comment:
this is the only one that is non standard, but as best as I can tell it evaluates the same